interp/src/interpreter/interpret_group.rs

```diff
@@ -287,16 +288,40 @@ fn interp_assignments<'a, I: Iterator<Item = &'a ir::Assignment>>(
         //if done signal is low and we haven't yet changed anything, means primitives are done,
         //time to evaluate sequential components
         if !is_signal_high(working_env.get(done_signal)) && !val_changed_flag {
-            working_env.do_tick();
+            let mut update_list: Vec<(RRC<ir::Port>, OutputValue)> = vec![];
+            //no need to do zero-assign check cuz this is run just once (?)
             for cell in cells.iter() {
                 if let Some(x) =
                     working_env.backing_env.cell_prim_map.borrow_mut().get_mut(
                         &(&cell.borrow() as &ir::Cell as *const ir::Cell),
                     )
                 {
-                    x.commit_updates()
+                    let new_vals = x.do_tick();
+                    for (port, val) in new_vals {
+                        //println!("updated port: {} w/ val {:?}", port.id, val);
+                        let port_ref = cell.borrow().find(port).unwrap();
+                        update_list.push((Rc::clone(&port_ref), val));
+                    }
+
+                    //call do_tick on each cell's primitive, and then
+                    //write the updates as done in eval_prims
+                    //specifically line 419, 448, 455 (create update list, push to it, env.update_val)
+                    //then call working_env.do_tick(), so everything is written to backing_env for us :=)
+                    //no need to worry about done values anymore; the done values will be written
+                    //to env, and primitives will remember their previous [done] state
+                    //no need to treat [done] special anymore.
                 }
             }
+            //put everything from update list into working_env, then
+            //working_env will put the IMM values (all values) into
+            //backing_env
+            for (port, val) in update_list {
+                working_env.update_val(&port.borrow(), val);
+            }
+            working_env.do_tick();
```

Comment:
Can we remove this now given the changes?

Reply:
I think that suits the next PR where we remove the `working_env`?

Reply:
Fair enough!